heatexchanger/server.py

```diff
@@ -76,10 +77,8 @@ def handleClose(self):
 
 if __name__ == "__main__":
     # TODO: uncomment to produce the initial CSM file before serving
-    m = Layer(3, 3)
-    m.cost = 1/m.Q
-    sol = m.localsolve()
-    LASTSOL[0] = ((3, 3), sol)
+    m, sol = designHX()
```

Comment:
If this changes then the bit above that solves during the actual serving needs to use designHX as well.

Reply:
I am still not sure which html address I have to call to get the serving to show up... I tried localhost:8000, but can't get it? Lmk what I have to do.

Reply:
Ugh, also the suggestion to make all subs arguments is so clunky. I guess we could just give designHX the arguments of Layer, and also the subs as an additional argument.

Reply:
Oh, to get the server to rebuild the model? you need to call it with the websocket format specified in hxesp.js, but unfortunately there's no GUI for that unless there's the right CSM file loaded. :-/

Reply:
Also also, designHX needs to return a model or accept a substitutions dictionary, so that we can apply the modified design parameters...there really was a reason I was creating new instances of the model here, I swear :p

Reply:
Also current master server.py returns exit status 10006 (stalled solution?) on my machine. No idea why, since the subs look ok. Odd.